tests/functional/spec/features/dynamic_answer_options/function_driven.js

```diff
@@ -0,0 +1,205 @@
+import DynamicCheckboxPage from "../../../generated_pages/dynamic_answer_options_function_driven_with_static_options/dynamic-checkbox.page";
+import DynamicRadioPage from "../../../generated_pages/dynamic_answer_options_function_driven_with_static_options/dynamic-radio.page";
+import DynamicDropdownPage from "../../../generated_pages/dynamic_answer_options_function_driven_with_static_options/dynamic-dropdown.page";
+import DynamicMutuallyExclusivePage from "../../../generated_pages/dynamic_answer_options_function_driven_with_static_options/dynamic-mutually-exclusive.page";
+import SubmitPage from "../../../generated_pages/dynamic_answer_options_function_driven_with_static_options/submit.page";
+
+const dropdownOptionValues = ["2020-12-28", "2020-12-29", "2020-12-30", "2020-12-31", "2021-01-01", "2021-01-02", "2021-01-03"];
+const dropdownOptionValuesWithStaticOption = [...dropdownOptionValues, "I did not work"];
+
+const testCases = [
+  {
+    schemaName: "test_dynamic_answer_options_function_driven_with_static_options.json",
+    answerOptionCount: 8,
+    dropdownOptionValues: dropdownOptionValuesWithStaticOption,
+  },
+  {
+    schemaName: "test_dynamic_answer_options_function_driven.json",
+    answerOptionCount: 7,
+    dropdownOptionValues: dropdownOptionValues,
+  },
+];
+
+const openQuestionnaireAndSetUp = (schema) => {
+  browser.openQuestionnaire(schema);
+  // Set reference date
+  $(ReferenceDatePage.day()).setValue("1");
+  $(ReferenceDatePage.month()).setValue("1");
+  $(ReferenceDatePage.year()).setValue("2021");
+  $(ReferenceDatePage.submit()).click();
+};
+
+testCases.forEach((testCase) => {
+  describe(`Feature: Dynamically generated answer options driven by a function (${testCase.schemaName})`, () => {
+    describe("Selecting/Deselecting", () => {
+      before("Open questionnaire", () => {
+        openQuestionnaireAndSetUp(testCase.schemaName);
+      });
+
+      describe("Given a dynamic answer options questionnaire and I am on a dynamic checkbox answer page", () => {
+        it("When I click a checkbox option, then the checkbox should be selected", () => {
+          for (let i = 0; i < testCase.answerOptionCount; i++) {
+            $(DynamicCheckboxPage.answerByIndex(i)).click();
+            expect($(DynamicCheckboxPage.answerByIndex(i)).isSelected()).to.be.true;
+          }
+        });
+
+        it("When I click a selected option, then it should be deselected", () => {
+          for (let i = 0; i < testCase.answerOptionCount; i++) {
+            $(DynamicCheckboxPage.answerByIndex(i)).click();
+            expect($(DynamicCheckboxPage.answerByIndex(i)).isSelected()).to.be.false;
+          }
+        });
+
+        it("When I submit the page, then I should be taken to the next page", () => {
+          $(DynamicCheckboxPage.submit()).click();
+          expect(browser.getUrl()).to.contain(DynamicRadioPage.pageName);
```

Comment:
Is this needed, isn't this just selecting and deselecting a checkbox? I see a few others below as well.

Reply:
This was to ensure the dynamic options were in fact rendered properly and that I am able to select/deselect them. If I were to only select the first option, we can't guarantee that the rest of the dynamic options were rendered properly.

Reply:
Still think it's a bit overkill, but not going to argue